crates/sol-types/src/types/data_type.rs

```diff
@@ -213,6 +213,50 @@ impl SolType for Address {
     }
 }
 
+/// Function - `function`
+pub struct Function;
+
+impl<T: Borrow<[u8; 24]>> Encodable<Function> for T {
```

Comment:
do we want `impl Encodable<Function> for (Address, Selector)` ?

Reply:
```
error[E0119]: conflicting implementations of trait `ty::Encodable<data_type::Function>` for type `(alloy_primitives::Address, [u8; 4])`
   --> crates/sol-types/src/types/data_type.rs:226:1
    |
219 | impl<T: Borrow<[u8; 24]>> Encodable<Function> for T {
    | --------------------------------------------------- first implementation here
...
226 | impl Encodable<Function> for (RustAddress, [u8; 4]) {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(alloy_primitives::Address, [u8; 4])`
    |
    = note: upstream crates may add a new impl of trait `std::borrow::Borrow<[u8; 24]>` for type `(alloy_primitives::Address, [u8; 4])` in future versions

```